src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -1468,6 +1468,27 @@ public static int Sign(float value)
             throw new ArithmeticException(SR.Arithmetic_NaN);
         }
 
+        // Helper functions for CopySign
+        // >= 0: returns 1
+        // < 0: returns -1
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
nit: While this is below the RyuJIT always inline size, and effectively has no impact in RyuJIT, that may not be true for all possible compilers that handle this code. Mono has different limits for example, and there are scenarios where code will only be inlined when its been attributed.

`MethodImplOptions` is one of the runtime attributes that has a specialized encoding in the method declaration, so we aren't wasting any space by specifying these and it can only help ensure our intent is followed on other runtimes.

Reply:
@tannergooding thanks for the explanation, I was not aware of that.

@jkotas @stephentoub in the past you have often asked me to remove similar annotations. What is our strategy for methods that get inlined by RyuJIT without the attribute being applied?

Reply:
The unnecessary annotations are code clutter. It is better to omit them if one can reasonably expect the method to be naturally inlined.

This method has 6 trivial IL instructions, so it is reasonable to expect that it is going to be inlined by any optimizing code generator.

Reply:
I don't think we can give that expectation to Mono at this time. There are a number of improvements still needed to their inliner before it will work as expected for an optimizing code generator. 

For example, mono only allows inlining methods that contain a constructor call if they are marked `AggressiveInlining` today and there was a discussion on that here: https://github.com/dotnet/runtime/pull/84660#discussion_r1164500106

There are many similar pessimizations and while this particular case might work, this is ultimately an internal helper method and the clutter is really not that bad. It's no worse than many of the other attributes we have in places and is easily skimmed over.

It also allows us to trivially identify methods we expect to be always inlined and can even improve things in RyuJIT in some cases since our "always inline" threshold isn't actually "always inline". It can still be skipped due to general budget limitations (for which `AggressiveInlining` can minimally increase) and can depend on the exact `InlinePolicy` being used. 

-- We are notably inconsistent in what this "always inline" threshold is and whether it is actually respected. For example the `DefaultPolicy` only skips for `12 bytes`, while `ALWAYS_INLINE_SIZE` is defined as `16`. We similarly make an observation of `CALLEE_BELOW_ALWAYS_INLINE_SIZE`, but never actually use that outside the CSV dump. We allow the `RandomPolicy` to ignore this as well (also soft coding it as `16`, rather than using the constant), we only use `m_CodeSize` as part of the general estimate for `ExtendedDefaultPolicy`, etc.

So I think there is likely still work needed to ensure that code size alone is functionally treated at least as equivalently as `AggressiveInlining` in RyuJIT.

Reply:
> that may not be true for all possible compilers that handle this code

A notable mention would be Unity IL2CPP which by default doesn't inline across types if `AggresiveInlining` isn't specified.

Reply:
I agree that there is a grey area where the inlining differs thresholds between Mono or between different RyuJIT heuristics.

If the method is a trivial wrapper for another method, a trivial property getter or trivial expression (e.g. no more than 3 ops), I think it is a reasonable to expect that the method will be inlined even without AggressiveInlining annotation.

Reply:
I logged https://github.com/dotnet/runtime/issues/94349 to track some of the potentially issues I noticed with the RyuJIT handling of the `always inline threshold`.

Reply:
> A notable mention would be Unity IL2CPP which by default doesn't inline across types if AggresiveInlining isn't specified.

It does not make sense to tell everyone to mark all trivial methods with AggresiveInlining to make them inline with IL2CPP. It is a problem for IL2CPP to solve.

IL2CPP depends on the C/C++ compiler for inlining. I believe that inlining in IL2CPP works just fine with link time codegen that is recommended for best performance. The problem only occurs for fast-build configurations without link time codegen or when link time codegen is not available. IL2CPP duplicates AggresiveInlining methods in every complication unit as a workaround for these cases. They can duplicate other trivial methods as well (if they are not doing it already). It is a tradeoff between build time and runtime performance.

Reply:
> I believe that inlining in IL2CPP works just fine with link time codegen that is recommended for best performance.

The issue would be that Unity doesn't enable LTCG/LTO in the default configuration, you can enable it in newer versions in the settings. Not sure how many people do so though.

Reply:
According to the docs, LTCG should be enabled in the master configuration that is recommended for shipping bits: https://docs.unity3d.com/ScriptReference/Il2CppCompilerConfiguration.html

Reply:
> in the master configuration that is recommended for shipping bits

That's right but the default configuration is Release, not Master.